includes/Admin.php

```diff
@@ -805,21 +807,97 @@ public function data_input_box_content( $post_or_order_object ) {
-				'name'      => "_wcpdf_{$document->slug}_number",
-			),
-			'date' => array(
-				'formatted' => $document->exists() && ! empty( $document->get_date() ) ? $document->get_date()->date_i18n( wc_date_format().' @ '.wc_time_format() ) : '',
-				'date'      => $document->exists() && ! empty( $document->get_date() ) ? $document->get_date()->date_i18n( 'Y-m-d' ) : date_i18n( 'Y-m-d' ),
-				'hour'      => $document->exists() && ! empty( $document->get_date() ) ? $document->get_date()->date_i18n( 'H' ) : date_i18n( 'H' ),
-				'minute'    => $document->exists() && ! empty( $document->get_date() ) ? $document->get_date()->date_i18n( 'i' ) : date_i18n( 'i' ),
+	/**
+	 * Returns the current values for the document data.
+	 * 
+	 * @param OrderDocument $document The document instance.
+	 * @param array $data The data to be processed.
+	 * 
+	 * @return array The current values for the document data.
+	 */
+	public function get_current_values_for_document_data( OrderDocument $document, array $data ): array {
+		$document_number_instance = $document->get_number();
+		$document_date_instance   = $document->get_date();
+		$current                  = array();
+		
+		if ( $document->exists() ) {
+			if ( ! empty( $document_number_instance ) ) {
+				$current['number'] = array(
+					'prefix' => array(
+						'value' => $document_number_instance->get_prefix() ?: null,
+						'name'  => "_wcpdf_{$document->slug}_number_prefix",	
+					),
+					'plain' => array(
+						'value' => $document_number_instance->get_plain() ?: null,
+						'name'  => "_wcpdf_{$document->slug}_number_plain",
+					),
+					'suffix' => array(
+						'value' => $document_number_instance->get_suffix() ?: null,
+						'name'  => "_wcpdf_{$document->slug}_number_suffix",
+					),
+					'padding' => array(
+						'value' => $document_number_instance->get_padding() ?: null,
+						'name'  => "_wcpdf_{$document->slug}_number_padding",
+					),
+					'formatted' => array(
+						'value' => $document_number_instance->get_formatted() ?: null,
+						'name'  => "_wcpdf_{$document->slug}_number_formatted",
+					),
+				);
+			}
+			
+			if ( ! empty( $document_date_instance ) ) {
+				$current['date'] = array(
+					'formatted' => $document_date_instance->date_i18n( wc_date_format().' @ '.wc_time_format() ) ?: null,
+					'date'      => $document_date_instance->date_i18n( 'Y-m-d' ) ?: date_i18n( 'Y-m-d' ),
+					'hour'      => $document_date_instance->date_i18n( 'H' ) ?: date_i18n( 'H' ),
+					'minute'    => $document_date_instance->date_i18n( 'i' ) ?: date_i18n( 'i' ),
+					'name'      => "_wcpdf_{$document->slug}_date",
+				);
+			}
+		} else {
+			$number_store    = $document->get_sequential_number_store();
+			$number_settings = $document->get_number_settings();
+			
+			$current['number'] = array(
+				'prefix' => array(
+					'value' => $number_settings['prefix'] ?: null,
```

Comment:
I'm seeing these warnings that appear above the documents data panel, when the document doesn't exist yet:

```
Warning: Undefined array key "prefix" in \wp-content\plugins\woocommerce-pdf-invoices-packing-slips\includes\Admin.php on line 864

Warning: Undefined array key "suffix" in \wp-content\plugins\woocommerce-pdf-invoices-packing-slips\includes\Admin.php on line 872

Warning: Undefined array key "padding" in \wp-content\plugins\woocommerce-pdf-invoices-packing-slips\includes\Admin.php on line 876
```

![image](https://github.com/user-attachments/assets/0f987df2-80da-4918-86d8-91a81453b324)